SQL/0000-00-00-schema.sql

```diff
@@ -2044,3 +2044,13 @@ CREATE TABLE `candidate_consent_history` (
   `EntryStaff` varchar(255) DEFAULT NULL,
   CONSTRAINT `PK_candidate_consent_history` PRIMARY KEY (`CandidateConsentHistoryID`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+
+-- A table to store the hashed value of passwords that were mistakenly recored
+-- in the `history` table. These passwords will not be allowed to be used as
+-- LORIS user passwords as they have potentially been compromised.
+CREATE TABLE `password_blacklist` (
+    `ID` integer unsigned NOT NULL AUTO_INCREMENT,
+    `password_hash` varchar(255) DEFAULT NULL,
```

Comment:
Is it a specific type of hash or can it be any hashing algorithm? (Presumably the former so that the blacklist can be compared against..) If it needs to be a specific hashing algorithm, the column should probably be named to reflect what it's expecting.

Reply:
As you noted below, the password algorithm can change so it's best not to hard-code a value here. PHP's default hashing algorithm at this time is bcrypt which stores an identifying prefix before the actual hash (`$2y`). `password_verify` takes care of identifying the algorithm, calculating the complexity, and checking that an input matches the hash.

Reply:
I don't think password_verify is suitable for the purposes of looking up a blacklist.